torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -565,6 +565,7 @@ XLAGraphExecutor::SyncTensorCollection XLAGraphExecutor::CollectSyncTensors(
 void XLAGraphExecutor::TensorCollectionBarrier(SyncTensorCollection* coll) {
   torch::lazy::LazyGraphExecutor::TensorCollectionBarrier(coll);
   // TODO(yeounoh) lock SPMD device
```

Comment:
This is great, so the PR#5018 would ensure that the live tensors end up having the correct SPMD:0 devices, then the locking can actually work here -- is this a correct understanding?